src/sysupdate/sysupdate.c

```diff
@@ -241,7 +241,7 @@ static int context_read_definitions(Context *c, const char* node) {
                         log_warning("As of v257, transfer definitions should have the '.transfer' extension.");
         }
 
-        if (c->n_transfers == 0) {
+        if ((enabled_only && c->n_transfers == 0) || c->n_transfers + c->n_disabled_transfers == 0) {
```

Comment:
hmm, would be easier to grok if this was:

```c
if (c->transfers + (relax_no_transfer_check ? 0 : c->n_disabled_transfers) == 0)) {
```

Reply:
or actually, i am confused. the way i read this this will result in errors in *more* cases, not less? what am i not grokking? /me is confused.

Reply:
I think you are confused as "relax_no_transfer_check" is a bad choice for the variable.
"need_enabled_transfers" would be a better choice for the variable name.
One class of verbs needs enabled transfers (update, check-new, ...). The other verbs will work with enabled and disabled (via features) transfers: "features" and "vacuum". They don't need any enabled transfer, they also do meaningful stuff on disabled transfers.